From bca9e2b904e6279d2e8cc74171a268e9171892eb Mon Sep 17 00:00:00 2001
From: PratikBhusal <PratikBhusal@users.noreply.github.com>
Date: Sun, 21 Oct 2018 14:02:23 -0500
Subject: [PATCH 1/2] Added image navigation wrapping

---
 commands.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/commands.c b/commands.c
index f685bc0..7b8d800 100644
--- a/commands.c
+++ b/commands.c
@@ -275,10 +275,10 @@ bool ci_navigate(arg_t n)
 	if (prefix > 0)
 		n *= prefix;
 	n += fileidx;
-	if (n < 0)
-		n = 0;
-	if (n >= filecnt)
-		n = filecnt - 1;
+	n = n % filecnt;
+	if (n < 0) {
+	    n += filecnt;
+	}
 
 	if (n != fileidx) {
 		load_image(n);
-- 
2.17.1


From 01e34f3b7ca6c856c759723cdec3928ab1622c10 Mon Sep 17 00:00:00 2001
From: PratikBhusal <PratikBhusal@users.noreply.github.com>
Date: Sun, 21 Oct 2018 14:09:53 -0500
Subject: [PATCH 2/2] Added documentation for ci_navigate() function

---
 commands.c | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/commands.c b/commands.c
index 7b8d800..9dd6e0b 100644
--- a/commands.c
+++ b/commands.c
@@ -270,6 +270,21 @@ bool cg_change_gamma(arg_t d)
 	}
 }
 
+/**
+ * @brief Navigate forwards/backwards to an image.
+ *
+ * @details alongside, the parameter n to indicate forwards/backwards movement,
+ * the function utilizes the following global variables: prefix, fileidx, and
+ * filecnt.
+ * @note The global variable "prefix" indicates how much to move by.
+ * @note The global variable "fileidx" indicates the 0-index location of the
+ *       image before movement.
+ * @note The global variable "filecnt" indicates the number of images accounted
+ *       for in the sxiv instance.
+ *
+ * @param n When n == 1, forwards movement. When n == -1, backwards movement.
+ * @return Loads different image if n != fileidx. Else, does nothing.
+ */
 bool ci_navigate(arg_t n)
 {
 	if (prefix > 0)
-- 
2.17.1

